src/main/webapp/app/modules/games/single-game/single-game.tsx

```diff
@@ -0,0 +1,128 @@
+import '../games.scss';
+
+import {RouteComponentProps} from "react-router-dom";
+import {connect} from "react-redux";
+import {IRootState} from "app/shared/reducers";
+import {getSingleGame} from "app/modules/games/games.reducer";
+import React, {useEffect, useState} from "react";
+import {Col, Container, DropdownItem, Row} from 'reactstrap';
+import {cleanText, formatCurrency} from "app/shared/util/render-utils";
+import {CartQuantityBox} from "app/modules/games/single-game/cart-quantity-box";
+import {CategoriesMechanics} from "app/modules/games/single-game/categories-mechanics";
+import {Carousel} from "app/modules/games/single-game/carousel";
+
+export interface ISingleGameProps extends StateProps, DispatchProps, RouteComponentProps<{ id: string }> {
+}
+
+export const SingleGame = (props: ISingleGameProps) => {
+  const [formattedCurrency, setFormattedCurrency] = useState("");
+  const [quantityDropdown, setQuantityDropdown] = useState([]);
+  const [desiredQuantity, setDesiredQuantity] = useState(1);
+  const [checkoutDisabled, setCheckoutDisabled] = useState(false);
+
+  const createQuantityDropdown = (quantity: number) => {
+    const items = [];
+    if (quantity <= 10) {
+      for (let i = 1; i <= quantity; i++) {
+        items.push(<DropdownItem onClick={() => {
+          setDesiredQuantity(i)
+        }}>{i}</DropdownItem>)
+      }
+    } else {
+      for (let i = 1; i <= 10; i++) {
+        items.push(<DropdownItem onClick={() => {
+          setDesiredQuantity(i)
+        }}>{i}</DropdownItem>)
+      }
+    }
+    return items;
+  }
+
+  useEffect(() => {
+    props.getSingleGame(props.match.params.id);
+  }, []);
+  const {
+    name, price, quantity, rulesUrl, largeImageUrl, description, categories, mechanics,
+    yearPublished, minimumAge, minimumPlayers, maximumPlayers, userRating
+  } = props.singleGame;
+
+  useEffect(() => {
+    setFormattedCurrency(formatCurrency(price))
+  }, [price]);
+
+  useEffect(() => {
+    setQuantityDropdown(createQuantityDropdown(quantity))
+    if (quantity === 0) {
+      setCheckoutDisabled(true)
+    } else {
+      setCheckoutDisabled(false)
+    }
+  }, [quantity]);
```

Comment:
Unclear if you need this. You can try it out and see.

Reply:
You're right - didn't need it 👍